src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioWorkspaceImpl.cs

```diff
@@ -2077,24 +2085,27 @@ internal void SetDependencyNodeTargetIdentifier(ProjectId projectId, string targ
                 (projectId, targetIdentifier));
         }
 
-        internal void RefreshProjectExistsUIContextForLanguage(string language)
+        internal async Task RefreshProjectExistsUIContextForLanguageAsync(string language, CancellationToken cancellationToken)
         {
-            // We must assert the call is on the foreground as setting UIContext.IsActive would otherwise do a COM RPC.
+            await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, cancellationToken);
+
             _foregroundObject.AssertIsForeground();
```

Comment:
Not really necessary anymore, unless we're really worried about SwitchToMainThreadAsync() doing strange things.

Reply:
yeah, that's fair.  i'm also SUPER paranoid :)